onchain/aiken/validators/settlement-queue.ak

```diff
@@ -397,39 +395,55 @@ validator mint(hub_oracle: ScriptHash) {
             withdrawals_root,
             start_time,
             end_time,
-            time_of_claimed_as_resolved: None,
-            operator: Some(operator_vkey),
+            resolution_claim: None,
           }
-        parsed_settlement_output_datum.data == expected_produced_node_data
+        expect
+          parsed_settlement_output_datum.data == expected_produced_node_data
+
+        // Done.
+        True
       }
-      RemoveSettlement {
+      RemoveSettlementNode {
         key_to_remove,
         removed_node_input_index,
         anchor_node_input_index,
       } -> {
-        expect Finite(valid_from) = validity_range.lower_bound.bound_type
-        expect Some(Output { datum: InlineDatum(removed_node_datum_data), .. }) =
-          node_in_outputs |> list.at(removed_node_input_index)
-        expect parsed_removed_node_datum: Datum = removed_node_datum_data
-        expect NodeData {
-          time_of_claimed_as_resolved: Some(
-            unwrapped_time_of_claimed_as_resolved,
-          ),
-          operator: Some(operator_vkey),
-          ..
-        }: NodeData = parsed_removed_node_datum.data
-        and {
+        // 1. The transaction must Remove a node from the `settlement_queue`.
+        //    Let `removed_settlement_node` be that node.
+        expect
           unordered.remove(
             Key(key_to_remove),
             removed_node_input_index,
             anchor_node_input_index,
             node_in_outputs,
             node_outputs,
             node_mint,
-          ),
-          operator_vkey |> utils.has_signed(extra_signatories),
-          unwrapped_time_of_claimed_as_resolved + settlement_resolution_duration <= valid_from,
-        }
+          )
+        //
+        expect Some(Output { datum: InlineDatum(removed_node_datum_data), .. }) =
+          node_in_outputs |> list.at(removed_node_input_index)
+
+        // 2. The `resolution_claim` of `removed_settlement_node` must not be empty.
+        expect parsed_removed_node_datum: Datum = removed_node_datum_data
+        expect NodeData {
+          resolution_claim: Some(ResolutionClaim {
+            resolution_time: unwrapped_time_of_claimed_as_resolved,
+            operator: operator_vkey,
+          }),
+          ..
+        }: NodeData = parsed_removed_node_datum.data
+
+        // 3. The transaction must be signed by the `operator` of `resolution_claim`.
+        expect operator_vkey |> utils.has_signed(extra_signatories)
+
+        // 4. The transaction's time-validity lower bound must match or exceed the
+        //    `resolution_time` of the `resolution_claim`.
+        expect Finite(valid_from) = validity_range.lower_bound.bound_type
+        expect
+          unwrapped_time_of_claimed_as_resolved + settlement_resolution_duration <= valid_from
```

Comment:
If I understood the logic flow correctly, `unwrapped_time_of_claimed_as_resolved` here already has `settlement_resolution_duration` included.
```suggestion
          unwrapped_time_of_claimed_as_resolved <= valid_from
```

Reply:
Yes, good catch.